ci/gpu/build.sh

```diff
@@ -139,6 +136,9 @@ if [[ -z "$PROJECT_FLASH" || "$PROJECT_FLASH" == "0" ]]; then
     ################################################################################
     # TEST - Run notebook tests
     ################################################################################
+    set +e -Eo pipefail
+    EXITCODE=0
+    trap "EXITCODE=1" ERR
 
     gpuci_logger "Notebook tests"
     ${WORKSPACE}/ci/gpu/test-notebooks.sh 2>&1 | tee nbtest.log
```

Comment:
The `test-notebooks.sh` script here isn't sourced, so I'm not sure the variables in it would be overwriting variables from the parent `build.sh` script. Does this sound accurate? What's the best way we can confirm this proposed solution?

Reply:
When a script is ran directly using its filepath, it is the equivalent of running `source`. To avoid a script being sourced, you'd have to start it from a separate bash shell, like `bash <script>`

Reply:
> When a script is ran directly using its filepath, it is the equivalent of running source. To avoid a script being sourced, you'd have to start it from a separate shell, like bash <script>

@dillon-cullinan, I don't think this statement is accurate. You can test it locally with the following two scripts:

`scr1.sh`:
```sh
#!/bin/bash

./scr2.sh
# source ./scr2.sh

echo $VAR1
```

`scr2.sh`:
```sh
#!/bin/bash

VAR1="some test string"
```

In `scr1.sh`, `$VAR1` is only available if you un-comment the `source` commmand.

Reply:
This answer on SO agrees with AJ: https://stackoverflow.com/a/8352939